apps/maestro/src/features/RegisterRemoteTokens/hooks/useRegisterRemoteStandardizedTokens.ts

```diff
@@ -46,26 +46,24 @@ export default function useRegisterRemoteStandardizedTokens(
 
   const { data: gasFees } = useEstimateGasFeeMultipleChainsQuery({
     destinationChainIds,
-    sourceChainId: sourceChainId ?? "0",
+    sourceChainId: sourceChain?.id ?? "0",
   });
 
   const multicallArgs = useMemo(() => {
     if (!tokenDeployment || !gasFees || tokenDeployment.kind !== "standardized")
       return [];
 
     return destinationChainIds.map((chainId, i) =>
-      encodeInterchainTokenServiceDeployAndRegisterRemoteStandardizedTokenData({
+      // encode InterchainTokenService DeployAndRegisterRemoteStandardizedTokenData
+      INTERCHAIN_TOKEN_FACTORY_ENCODERS.deployRemoteInterchainToken.data({
         salt: tokenDeployment.salt,
-        name: tokenDeployment.tokenName,
-        symbol: tokenDeployment.tokenSymbol,
-        decimals: tokenDeployment.tokenDecimals,
+        originalChainName: sourceChain?.name ?? "",
         distributor: "0x", // remote tokens cannot be minted, so the distributor must be 0x
```

Comment:
if distributor is not selected as `0` by user, we should check if it matches the distributor set on token on the current chain via `isDistributor`. For security, the factory doesn't allow changing the distributor to any address for future deployments.